components/tournament/tournament.lua

```diff
@@ -0,0 +1,170 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Tournament
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local DateExt = require('Module:Date/Ext')
+local Lpdb = require('Module:Lpdb')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local Tier = require('Module:Tier/Utils')
+
+local Tournaments = {}
+
+---@enum TournamentPhase
+local TOURNAMENT_PHASE = {
+	UPCOMING = 'UPCOMING',
+	ONGOING = 'ONGOING',
+	FINISHED = 'FINISHED',
+}
+
+---@class StandardTournament
+---@field displayName string
+---@field pageName string
+---@field startDate {year: integer, month: integer?, day: integer?, timestamp: integer?}?
+---@field endDate {year: integer, month: integer?, day: integer?, timestamp: integer?}?
+---@field liquipediaTier string
+---@field liquipediaTierType string
+---@field region string
+---@field featured boolean
+---@field status string?
+---@field phase TournamentPhase
+
+---@param conditions ConditionTree?
+---@return StandardTournament[]
+function Tournaments.getAllTournaments(conditions)
+	local tournaments = {}
+	Lpdb.executeMassQuery(
+		'tournament',
+		{
+			conditions = conditions and conditions:toString() or nil,
+			order = 'sortdate desc',
+		},
+		function(record)
+			local tournament = Tournaments.tournamentFromRecord(
+				record,
+				Tournaments.makeFeaturedFunction()
+			)
+			table.insert(tournaments, tournament)
+		end
+	)
+	return tournaments
+end
+
+---@param record tournament
+---@param recordIsFeatured function
+---@return StandardTournament
+function Tournaments.tournamentFromRecord(record, recordIsFeatured)
+	local startDate = Tournaments.parseDateRecord(record.startdate)
+	local endDate = Tournaments.parseDateRecord(record.sortdate or record.enddate)
+
+	local tournament = {
+		displayName = Logic.emptyOr(record.tickername, record.name) or record.pagename:gsub('_', ' '),
+		fullName = record.name,
+		pageName = record.pageName,
+		startDate = startDate,
+		endDate = endDate,
+		liquipediaTier = Tier.toIdentifier(record.liquipediatier),
+		liquipediaTierType = Tier.toIdentifier(record.liquipediatiertype),
+		region = record.region,
+		status = record.status,
+		featured = recordIsFeatured(record),
+	}
+
+	tournament.phase = Tournaments.calculatePhase(tournament)
+
+	return tournament
+end
+
+---@param tournament StandardTournament
+---@return TournamentPhase
+function Tournaments.calculatePhase(tournament)
+	if tournament.status == 'finished' then
+		return TOURNAMENT_PHASE.FINISHED
+	end
+	if not tournament.startDate then
+		return TOURNAMENT_PHASE.UPCOMING
+	end
+	if DateExt.getCurrentTimestamp() < tournament.startDate.timestamp then
+		return TOURNAMENT_PHASE.UPCOMING
+	end
+	if not tournament.endDate.timestamp then
+		return TOURNAMENT_PHASE.ONGOING
+	end
+	if DateExt.getCurrentTimestamp() < tournament.endDate.timestamp then
```

Comment:
```suggestion
	if DateExt.getCurrentTimestamp() < (tournament.endDate.timestamp + 24 * 60 * 60) then
```
so the events are not finished during their last day of play
e.g. 1-day events would never be listed as ongoing otherwise
(dates in tournament context are always set as T00:00:00 iirc)